sdcm/utils/remote_logger.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import subprocess
+import abc
+from multiprocessing import Process, Event
+from datetime import datetime
+from sdcm import wait
+from sdcm.sct_events import raise_event_on_failure
+from sdcm.remote import CommandRunner
+
+
+class LoggerBase:
+    def __init__(self, node, target_log_file: str):
+        self._node = node
+        self._target_log_file = target_log_file
+        self._log = logging.getLogger(self.__class__.__name__)
+
+    @abc.abstractmethod
+    def start(self):
+        raise NotImplementedError("To be overridden")
+
+    @abc.abstractmethod
+    def stop(self, timeout=None):
+        raise NotImplementedError("To be overridden")
+
+
+class SSHLoggerBase(LoggerBase):
+    _logger_cmd = None
+    _retrieve_message = "Reading Scylla logs from {since}"
+
+    def __init__(self, node, target_log_file: str):
+        super().__init__(node, target_log_file)
+        self._termination_event = Event()
+        self._remoter = None
+        self._remoter_params = node.remoter.get_instance_parameters()
+        self._child_process = Process(target=self._journal_thread, daemon=True)
+        if self._logger_cmd is None:
```

Comment:
where and how it should be overridden? Generally in OOP we override inherited methods not class attributes...

Reply:
Define in the Base class if you intend this to be overriden 
```python
    @property
    @abstractmethod
    def _logger_cmd(self):
          pass
```